GameServer/Logic/GameObjects/AzirTurret.cs

```diff
@@ -0,0 +1,209 @@
+        {
+            var blueTeam = new List<Champion>();
+            var purpTeam = new List<Champion>();
+            foreach (var player in _game.GetMap().GetAllChampionsFromTeam(TeamId.TEAM_BLUE))
+            {
+                blueTeam.Add(player);
+            }
+
+            foreach (var player in _game.GetMap().GetAllChampionsFromTeam(TeamId.TEAM_PURPLE))
+            {
+                purpTeam.Add(player);
+            }
+            if (getTeam() == TeamId.TEAM_BLUE)
+                return purpTeam.Count;
+            else
+                return blueTeam.Count;
+        }
+
+        public void buildTurret()
+        {
+            Inibin inibin;
+            if (!RAFManager.getInstance().readInibin("DATA/Characters/" + model + "/" + model + ".inibin", out inibin))
+            {
+                Logger.LogCoreError("couldn't find turret stats for " + model);
+                return;
+            }
+
+            stats.HealthPoints.BaseValue = inibin.getFloatValue("Data", "BaseHP");
+            stats.CurrentHealth = stats.HealthPoints.Total;
+            stats.ManaPoints.BaseValue = inibin.getFloatValue("Data", "BaseMP");
+            stats.CurrentMana = stats.ManaPoints.Total;
+            stats.AttackDamage.BaseValue = inibin.getFloatValue("DATA", "BaseDamage");
+            stats.Range.BaseValue = TURRET_RANGE;//inibin.getFloatValue("DATA", "AttackRange");
+            stats.MoveSpeed.BaseValue = inibin.getFloatValue("DATA", "MoveSpeed");
+            stats.Armor.BaseValue = inibin.getFloatValue("DATA", "Armor");
+            stats.MagicResist.BaseValue = inibin.getFloatValue("DATA", "SpellBlock");
+            stats.HealthRegeneration.BaseValue = inibin.getFloatValue("DATA", "BaseStaticHPRegen");
+            stats.ManaRegeneration.BaseValue = inibin.getFloatValue("DATA", "BaseStaticMPRegen");
+            stats.AttackSpeedFlat = 0.625f / (1 + inibin.getFloatValue("DATA", "AttackDelayOffsetPercent"));
+
+            stats.HealthPerLevel = inibin.getFloatValue("DATA", "HPPerLevel");
+            stats.ManaPerLevel = inibin.getFloatValue("DATA", "MPPerLevel");
+            stats.AdPerLevel = inibin.getFloatValue("DATA", "DamagePerLevel");
+            stats.ArmorPerLevel = inibin.getFloatValue("DATA", "ArmorPerLevel");
+            stats.MagicResistPerLevel = inibin.getFloatValue("DATA", "SpellBlockPerLevel");
+            stats.HealthRegenerationPerLevel = inibin.getFloatValue("DATA", "HPRegenPerLevel");
+            stats.ManaRegenerationPerLevel = inibin.getFloatValue("DATA", "MPRegenPerLevel");
+            stats.GrowthAttackSpeed = inibin.getFloatValue("DATA", "AttackSpeedPerLevel");
+
+            setMelee(inibin.getBoolValue("DATA", "IsMelee"));
+            setCollisionRadius(inibin.getIntValue("DATA", "PathfindingCollisionRadius"));
+
+            Inibin autoAttack;
+            if (!RAFManager.getInstance().readInibin("DATA/Characters/" + model + "/" + model + "BasicAttack.inibin", out autoAttack))
```

Comment:
I'd suggest you add the following method to `RAFManager` and just call that:
```CS
public Inibin GetAutoAttack(string model)
{
    Inibin autoAttack
    var autoAttackPaths = string[]
    {
        string.Format("DATA/Characters/{0}/{0}BasicAttack.inibin", model),
        string.Format("DATA/Characters/{0}/Spells/{0}BasicAttack.inibin", model),
        string.Format("DATA/Spells/{0}BasicAttack.inibin", model)
    }
    foreach(var path in autoAttackPaths)
    {
        if(readInibin(path, out autoAttack))
        {
            break;
        }
    }
    if(autoAttack == null)
    {
        Logger.LogCoreError(string.Format("Couldn't find turret auto-attack data for {0}", model))
    }
    return autoAttack
}
```

All reading from RAFs should be kept in the same class, will help later when converting to json. That should also be a bit better implementation of the functionality.